File: simpleast-core/src/main/java/com/agarron/simpleast_core/node/TextNode.java
package com.agarron.simpleast_core.node;


import android.content.Context;
import android.text.SpannableStringBuilder;

import com.agarron.simpleast_core.renderer.SpannableRenderableNode;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class TextNode extends SpannableRenderableNode {

    public final String content;

    public TextNode(final String content) {
        this.content = content;
    }

  @Override
  public void render(@NotNull SpannableStringBuilder builder, @Nullable Context context) {
    builder.append(content);
  }
}

File: simpleast-core/src/main/java/com/agarron/simpleast_core/node/TextNode.kt
package com.agarron.simpleast_core.node


import android.content.Context
import android.text.SpannableStringBuilder

import com.agarron.simpleast_core.renderer.SpannableRenderableNode

/**
 * Node representing simple text.
 */
class TextNode(val content: String) : SpannableRenderableNode() {

  override fun render(builder: SpannableStringBuilder, context: Context?) {
    builder.append(content)
  }
}
